---
layout: page
title: Practical Hisat2
summary: "Use of  Hisat2 for differential  analisys"
---

## Introduction

    
HISAT2 is a fast and sensitive alignment program for mapping next-generation sequencing reads (both DNA and RNA) to a population of human genomes (as well as to a single reference genome). Based on an extension of BWT for graphs [Sir√©n et al. 2014], we designed and implemented a graph FM index (GFM), an original approach and its first implementation to the best of our knowledge. In addition to using one global GFM index that represents a population of human genomes, HISAT2 uses a large set of small GFM indexes that collectively cover the whole genome (each index representing a genomic region of 56 Kbp, with 55,000 indexes needed to cover the human population). These small indexes (called local indexes), combined with several alignment strategies, enable rapid and accurate alignment of sequencing reads. This new indexing scheme is called a Hierarchical Graph FM index (HGFM).

This brief tutorial will explain how you can get started using Hisat2 to quantify your RNA-seq data.  Our computational power are not enought for elaborate a rela case so we use only a part of the data. We suggest to try at home with the  complete stock of data.
Computer avaible are not  born for  this purpose (oly 4G of RAM, It is not enough for the  new  operation systems) we need more patience






    # map each sample using 1 threads
    mkdir map
    hisat2 -p 1 --rg-id=ERR188044 --rg SM:ERR188044  --rg PL:ILLUMINA --rg PU:aaaaaaa -x chrX_data/indexes/chrX_tran --dta --rna-strandness RF -1 chrX_data/samples/ERR188044_chrX_1.fastq.gz -2 chrX_data/samples/ERR188044_chrX_2.fastq.gz -S map/ERR188044_chrX.sam
    hisat2 -p 1 --rg-id=ERR188104 --rg SM:ERR188104  --rg PL:ILLUMINA --rg PU:aaaaaaa -x chrX_data/indexes/chrX_tran --dta --rna-strandness RF -1 chrX_data/samples/ERR188104_chrX_1.fastq.gz -2 chrX_data/samples/ERR188104_chrX_2.fastq.gz -S map/ERR188104_chrX.sam
    hisat2 -p 1 -x chrX_data/indexes/chrX_tran -1 chrX_data/samples/ERR188234_chrX_1.fastq.gz -2 chrX_data/samples/ERR188234_chrX_2.fastq.gz -S map/ERR188234_chrX.sam
    hisat2 -p 1 -x chrX_data/indexes/chrX_tran   -1 chrX_data/samples/ERR188245_chrX_1.fastq.gz -2 chrX_data/samples/ERR188245_chrX_2.fastq.gz -S map/ERR188245_chrX.sam
    hisat2 -p 1 -x chrX_data/indexes/chrX_tran -1 chrX_data/samples/ERR188257_chrX_1.fastq.gz -2 chrX_data/samples/ERR188257_chrX_2.fastq.gz -S map/ERR188257_chrX.sam
    hisat2 -p 1 -x chrX_data/indexes/chrX_tran-1 chrX_data/samples/ERR188273_chrX_1.fastq.gz -2 chrX_data/samples/ERR188273_chrX_2.fastq.gz -S map/ERR188273_chrX.sam
    hisat2 -p 1 -x chrX_data/indexes/chrX_tran -1 chrX_data/samples/ERR188337_chrX_1.fastq.gz -2 chrX_data/samples/ERR188337_chrX_2.fastq.gz -S map/ERR188337_chrX.sam
    hisat2 -p 1 -x chrX_data/indexes/chrX_tran -1 chrX_data/samples/ERR188383_chrX_1.fastq.gz -2 chrX_data/samples/ERR188383_chrX_2.fastq.gz -S map/ERR188383_chrX.sam
    hisat2 -p 1 -x chrX_data/indexes/chrX_tran -1 chrX_data/samples/ERR188401_chrX_1.fastq.gz -2 chrX_data/samples/ERR188401_chrX_2.fastq.gz -S map/ERR188401_chrX.sam
    hisat2 -p 1 -x chrX_data/indexes/chrX_tran -1 chrX_data/samples/ERR188428_chrX_1.fastq.gz -2 chrX_data/samples/ERR188428_chrX_2.fastq.gz -S map/ERR188428_chrX.sam
    hisat2 -p 1 -x chrX_data/indexes/chrX_tran -1 chrX_data/samples/ERR188454_chrX_1.fastq.gz -2 chrX_data/samples/ERR188454_chrX_2.fastq.gz -S map/ERR188454_chrX.sam
    hisat2 -p 1 -x chrX_data/indexes/chrX_tran -1 chrX_data/samples/ERR204916_chrX_1.fastq.gz -2 chrX_data/samples/ERR204916_chrX_2.fastq.gz -S map/ERR204916_chrX.sam
    
    # sort all
    samtools sort -@ 1 -o map/ERR188044_chrX.bam map/ERR188044_chrX.sam
    samtools sort -@ 1 -o map/ERR188104_chrX.bam map/ERR188104_chrX.sam
    samtools sort -@ 1 -o map/ERR188234_chrX.bam map/ERR188234_chrX.sam
    samtools sort -@ 1 -o map/ERR188245_chrX.bam map/ERR188245_chrX.sam
    samtools sort -@ 1 -o map/ERR188257_chrX.bam map/ERR188257_chrX.sam
    samtools sort -@ 1 -o map/ERR188273_chrX.bam map/ERR188273_chrX.sam
    samtools sort -@ 1 -o map/ERR188337_chrX.bam map/ERR188337_chrX.sam
    samtools sort -@ 1 -o map/ERR188383_chrX.bam map/ERR188383_chrX.sam
    samtools sort -@ 1 -o map/ERR188401_chrX.bam map/ERR188401_chrX.sam
    samtools sort -@ 1 -o map/ERR188428_chrX.bam map/ERR188428_chrX.sam
    samtools sort -@ 1 -o map/ERR188454_chrX.bam map/ERR188454_chrX.sam
    samtools sort -@ 1 -o map/ERR204916_chrX.bam map/ERR204916_chrX.sam
    
    # remove SAM files
    rm map/*.sam
    mkdir assembly
    
    # create assembly per sample using 8 threads
    stringtie map/ERR188044_chrX.bam -l ERR188044 -p 1 -G chrX_data/genes/chrX.gtf -o assembly/ERR188044_chrX.gtf
    stringtie map/ERR188104_chrX.bam -l ERR188104 -p 1 -G chrX_data/genes/chrX.gtf -o assembly/ERR188104_chrX.gtf
    stringtie map/ERR188234_chrX.bam -l ERR188234 -p 1 -G chrX_data/genes/chrX.gtf -o assembly/ERR188234_chrX.gtf
    stringtie map/ERR188245_chrX.bam -l ERR188245 -p 1 -G chrX_data/genes/chrX.gtf -o assembly/ERR188245_chrX.gtf
    stringtie map/ERR188257_chrX.bam -l ERR188257 -p 1 -G chrX_data/genes/chrX.gtf -o assembly/ERR188257_chrX.gtf
    stringtie map/ERR188273_chrX.bam -l ERR188273 -p 1 -G chrX_data/genes/chrX.gtf -o assembly/ERR188273_chrX.gtf
    stringtie map/ERR188337_chrX.bam -l ERR188337 -p 1 -G chrX_data/genes/chrX.gtf -o assembly/ERR188337_chrX.gtf
    stringtie map/ERR188383_chrX.bam -l ERR188383 -p 1 -G chrX_data/genes/chrX.gtf -o assembly/ERR188383_chrX.gtf
    stringtie map/ERR188401_chrX.bam -l ERR188401 -p 1 -G chrX_data/genes/chrX.gtf -o assembly/ERR188401_chrX.gtf
    stringtie map/ERR188428_chrX.bam -l ERR188428 -p 1 -G chrX_data/genes/chrX.gtf -o assembly/ERR188428_chrX.gtf
    stringtie map/ERR188454_chrX.bam -l ERR188454 -p 1 -G chrX_data/genes/chrX.gtf -o assembly/ERR188454_chrX.gtf
    stringtie map/ERR204916_chrX.bam -l ERR204916 -p 1 -G chrX_data/genes/chrX.gtf -o assembly/ERR204916_chrX.gtf
    # before merging we need to modify mergelist.txt
    # this is because I created a new directory to store the results
    # the modified mergelist.txt should look like this
    cat chrX_data/mergelist.txt
    assembly/ERR188044_chrX.gtf
    assembly/ERR188104_chrX.gtf
    assembly/ERR188234_chrX.gtf
    assembly/ERR188245_chrX.gtf
    assembly/ERR188257_chrX.gtf
    assembly/ERR188273_chrX.gtf
    assembly/ERR188337_chrX.gtf
    assembly/ERR188383_chrX.gtf
    assembly/ERR188401_chrX.gtf
    assembly/ERR188428_chrX.gtf
    assembly/ERR188454_chrX.gtf
    assembly/ERR204916_chrX.gtf


    # merge all transcripts from the different samples
    stringtie --merge -p 1 -G chrX_data/genes/chrX.gtf -o stringtie_merged.gtf chrX_data/mergelist.txt
    # check out the transcripts
    cat stringtie_merged.gtf | head
    # stringtie --merge -p 8 -G chrX_data/genes/chrX.gtf -o stringtie_merged.gtf chrX_data/mergelist.txt
    # StringTie version 1.3.3b
    chrX    StringTie       transcript      322514  323718  1000    .       .       gene_id "MSTRG.1"; transcript_id "MSTRG.1.1"; 
    chrX    StringTie       exon    322514  323718  1000    .       .       gene_id "MSTRG.1"; transcript_id "MSTRG.1.1"; exon_number "1"; 
    chrX    StringTie       transcript      319145  321319  1000    +       .       gene_id "MSTRG.2"; transcript_id "NR_027232"; gene_name "LINC00685"; ref_gene_id "NR_027232"; 
    chrX    StringTie       exon    319145  321319  1000    +       .       gene_id "MSTRG.2"; transcript_id "NR_027232"; exon_number "1"; gene_name "LINC00685"; ref_gene_id "NR_027232"; 
    chrX    StringTie       transcript      319145  321319  1000    +       .       gene_id "MSTRG.2"; transcript_id "NR_027231"; gene_name "LINC00685"; ref_gene_id "NR_027231"; 
    chrX    StringTie       exon    319145  319551  1000    +       .       gene_id "MSTRG.2"; transcript_id "NR_027231"; exon_number "1"; gene_name "LINC00685"; ref_gene_id "NR_027231"; 
    chrX    StringTie       exon    320208  321319  1000    +       .       gene_id "MSTRG.2"; transcript_id "NR_027231"; exon_number "2"; gene_name "LINC00685"; ref_gene_id "NR_027231"; 
    chrX    StringTie       transcript      304750  318701  1000    -       .       gene_id "MSTRG.3"; transcript_id "MSTRG.3.1";
    
    # how many transcripts?
    cat stringtie_merged.gtf | grep -v "^#" | awk '$3=="transcript" {print}' | wc -l
        3491